### PR TITLE
Make it possible to use `FilePublisher` without running a nameserver

### PR DESCRIPTION
**Pull request by pnuu**
This is part of my slowly ongoing plan on using Trollflow2 in a container on operations. For our (FMI) use case it might be necessary to use direct port connections between different parts of the whole chain (Trollmoves Server -> Trollmoves Client -> segment gatherer -> Trollflow2 -> Georest -> Geoserver) without any of the steps using Posttroll `nameservers`s.

The default (below) usage will still require a Posttroll `nameserver`, and the publisher will default to use the one on `localhost'. By default the publish port is automatically selected
```yaml
workers:
  - fun: !!python/object:trollflow2.plugins.FilePublisher {}
```

To use direct port communications without connecting to a `nameserver`:
```yaml
workers:
  - fun: !!python/object:trollflow2.plugins.FilePublisher {port=40000, nameserver: null}
```

As with the default use, the messages will be published on the localhost in a port either given by the user or selected automatically by Posttroll, but with the feature added here the `nameserver` will not be needed if the port is known.

 - [x] Tests added  
 - [x] Tests passed 
 - [x] Passes ``flake8 trollflow2`` 
 - [ ] Fully documented

### CODE REVIEW

File: trollflow2/plugins/__init__.py [hunk 1]
```diff
@@ -279,10 +279,13 @@ def __setstate__(self, kwargs):
...
-        self.nameservers = kwargs.get('nameservers', None)
-        self.pub = NoisyPublisher('l2processor', port=self.port,
-                                  nameservers=self.nameservers)
-        self.pub.start()
+        self.nameservers = kwargs.get('nameservers', "")
+        if self.nameservers is None:
+            self.pub = Publisher("tcp://*:" + str(self.port), "l2processor")
+        else:
+            self.pub = NoisyPublisher('l2processor', port=self.port,
+                                      nameservers=self.nameservers)
```
Comment (mraspaud) [inline trollflow2/plugins/__init__.py:287]: I think @adybbroe just stumbled on a problem with nameservers set to None, right?

**Review by mraspaud — COMMENTED** [review]
Looks good! maybe some refactoring possibilities?

File: trollflow2/tests/test_trollflow2.py [hunk 2]
```diff
@@ -1285,63 +1285,67 @@ def test_filepublisher_with_compose(self):
...
-            for area in job['product_list']['product_list']['areas']:
-                for _prod in job['product_list']['product_list']['areas'][area]:
-                    # Skip calls to __str__
-                    if 'call().__str__()' != str(message.mock_calls[i]):
-                        self.assertTrue(topics[i] in str(message.mock_calls[i]))
-                        i += 1
-            self.assertEqual(message.call_args[0][2]['processing_center'], 'SMHI')
+        with mock.patch('trollflow2.plugins.Message') as message:
+            with mock.patch('trollflow2.plugins.NoisyPublisher'):
+                with mock.patch('trollflow2.plugins.Publisher'):
```
[the remark this answers is not in the document]
Comment (pnuu) [reply trollflow2/tests/test_trollflow2.py]: I think using the `mock.patch.multiple()` already makes this much cleaner. Creating custom context managers would make less readable.

**Review by mraspaud — APPROVED** [review]
LGTM

**Pull request merged**